crates/ruff/src/rules/pylint/rules/compare_to_empty_string.rs

```diff
@@ -0,0 +1,42 @@
+use itertools::Itertools;
+use ruff_macros::{derive_message_formats, violation};
+use rustpython_parser::ast::{Cmpop, Expr, ExprKind, Located};
+
+use ruff_python_ast::helpers::unparse_constant;
+use ruff_python_ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+#[violation]
+pub struct CompareToEmptyString;
+
+impl Violation for CompareToEmptyString {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("todo")
+    }
+}
+
+pub fn compare_to_empty_string(
+    checker: &mut Checker,
+    left: &Expr,
+    ops: &[Cmpop],
+    comparators: &[Expr],
+) {
+    for ((left, rhs), op) in std::iter::once(left)
+        .chain(comparators.iter())
+        .tuple_windows::<(&Located<_>, &Located<_>)>()
+        .zip(ops)
+    {
+        if matches!(op, Cmpop::Eq | Cmpop::NotEq) {
+            if let ExprKind::Constant { value: v, .. } = &rhs.node {
+                let k = unparse_constant(v, checker.stylist);
+                if k == "\"\"" || k == "''" {
```

Comment:
My recommendation would be to instead `match` on the `value`, and check for `value.is_empty()` in `Constant::Str(value)`. Would that have any problems? (It could also catch cases in which the string had a prefix, like `u`, which this would miss.)

Reply:
Separately: does Pylint catch `x == b""`? Or `x == f""`?

Reply:
I think matching on the value definitely makes sense.

I just tested the latest stable release of pylint and it does not catch the b"" or f"" cases, it only shows the lint warning for `""` without any specifiers. Pylint also shows the lint warning for `x is ""` and `x is not ""` which seems sensible as well.